src/components/SearchBar/SearchBar.tsx

```diff
@@ -1,159 +1,53 @@
+        ariaLabel,
+        ariaLabelledBy,
         blockName,
-        searchBarId,
-        buttonId,
-        searchBarAriaLabel,
-        searchBarAriaLabelledBy,
-        dropdownId,
-        options,
-        dropdownAriaLabel,
-        selectedField,
-        searchValue,
-        placeholderText,
-        helperVariant,
-        hasError,
-        errorMessage,
-        selectBlurHandler,
-        searchSubmitHandler,
-        selectChangeHandler,
-        searchChangeHandler,
+        children,
+        id,
+        onSubmit,
     } = props;
 
-    if (options) {
-        if (
-            !(
-                dropdownId &&
-                dropdownAriaLabel &&
-                selectBlurHandler &&
-                selectChangeHandler
-            )
-        ) {
-            throw new Error(`If dropdownOptions are passed, dropdownId (currently ${dropdownId}),
-      dropdownAriaLabel (currently ${dropdownAriaLabel}), selectChangeHandler (currently ${selectChangeHandler}),
-      and selectBlurHandler (currently ${selectBlurHandler}) must also be passed`);
-        }
-    }
-
-    if (!searchBarAriaLabel && !searchBarAriaLabelledBy) {
-        throw new Error(
-            "Either searchBarAriaLabel and searchBarAriaLabelledBy must be passed"
-        );
-    }
-
     let modifiers = [];
-    if (hasError) {
-        if (!errorMessage) {
-            throw new Error(
-                "If there is an error, there must also be an error message"
-            );
-        }
-        modifiers.push("error");
-    }
-
-    console.log("beepbeep:" + typeof options);
 
     let searchbar__base_class = "search-bar";
 
-    let textfieldProps = {
-        id: searchBarId + "-input-textfield",
-        ariaLabelledBy: buttonId,
-        onChange: searchChangeHandler,
-        isRequired: true,
-        blockName: searchbar__base_class,
-        placeholderText:
-            hasError && helperVariant ? errorMessage : placeholderText,
-        value: searchValue,
-        modifiers: modifiers,
-    };
-
-    let buttonProps: ButtonOptions = {
-        id: buttonId,
-        onClick: searchSubmitHandler,
-        blockName: searchbar__base_class,
-        content: <>Search</>,
-        buttonType: ButtonTypes.Primary,
-        iconPosition: ButtonIconPositions.Left,
-        iconName: "search_small",
-        iconDecorative: true,
-    };
-
     return (
         <form
-            className={`${bem(
-                searchbar__base_class,
-                modifiers,
-                blockName
-            )} ${bem(searchbar__base_class, modifiers)}`}
-            id={searchBarId}
+            className={bem(searchbar__base_class, modifiers, blockName)}
+            id={id}
             role="search"
-            aria-label={searchBarAriaLabel}
-            aria-labelledby={searchBarAriaLabelledBy}
+            aria-label={ariaLabel}
+            aria-labelledby={ariaLabelledBy}
+            onSubmit={props.onSubmit}
+            {...props.attributes}
```

Comment:
Assign these to consts in the L27 block so that we don't have to call `props` directly.